### PR TITLE
Patch: Fix patches that incorrectly depend on cluster member being leader

### PR DESCRIPTION
**Pull request by roosterfish**
This PR is a follow up on https://github.com/canonical/lxd/pull/13162 to fix the leader election also for `patchStorageRenameCustomISOBlockVolumes` and `patchStorageUnsetInvalidBlockSettings`.

A new function `selectedPatchClusterMember` is added which offers a deterministic approach to select a single member across the cluster which is eligible to apply the patch. For example this is necessary for the patch `patchStorageRenameCustomISOBlockVolumes` which interferes with the storage subsystem.

The patch `patchStorageUnsetInvalidBlockSettings` is superseded by a single idempotent query that can now run multiple times on all cluster members.

### CODE REVIEW

**Comment by roosterfish** [comment]
@tomponline do you want me to fix it also for `patchDBNodesAutoInc` or should we leave it there for historical reasons since it's not affected by the bug? 

I put the reworked patches alongside there parents to keep them together and not at the end of the file.

**Comment by tomponline** [comment]
> @tomponline do you want me to fix it also for `patchDBNodesAutoInc` or should we leave it there for historical reasons since it's not affected by the bug?

No, lets leave that one well alone.

**Review by tomponline — COMMENTED** [review]
The patches var definition says:

> Only append to the patches list, never remove entries and never re-order them.

File: lxd/patches.go [hunk 1]
```diff
@@ -950,6 +951,115 @@ func patchStorageRenameCustomISOBlockVolumes(name string, d *Daemon) error {
...
+			if hasVol {
+				continue
+			}
+
+			// We need to use ContentTypeBlock here in order for the driver to figure out the correct (old) location.
+			oldVol := storageDrivers.NewVolume(p.Driver(), p.Name(), storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeBlock, project.StorageVolume(vol.Project, vol.Name), nil, nil)
+
+			err = p.Driver().RenameVolume(oldVol, fmt.Sprintf("%s.iso", oldVol.Name()), nil)
+			if err != nil {
+				return fmt.Errorf("Failed renaming volume: %w", err)
```
Comment (tomponline) [inline lxd/patches.go]: Would be useful to include the volume info here into the error in case it fails.

**Comment by roosterfish** [comment]
> The patches var definition says:
> 
> > Only append to the patches list, never remove entries and never re-order them.

Mhh ok haven't seen this, it felt wrong doing it but I thought if the original patch gets emptied out, and its replacement is executed at the end of the list, maybe it might cause side effects with other patches.

File: lxd/patches.go [hunk 2]
```diff
@@ -950,6 +951,115 @@ func patchStorageRenameCustomISOBlockVolumes(name string, d *Daemon) error {
...
+			if vol.ContentType != dbCluster.StoragePoolVolumeContentTypeNameISO {
+				continue
+			}
+
+			// The existing volume using the actual *.iso suffix has ContentTypeISO.
+			existingVol := storageDrivers.NewVolume(p.Driver(), p.Name(), storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeISO, project.StorageVolume(vol.Project, vol.Name), nil, nil)
+
+			hasVol, err := p.Driver().HasVolume(existingVol)
+			if err != nil {
+				return err
```
[the remark this answers is not in the document]
Comment (roosterfish) [reply lxd/patches.go]: I agree, fixed.

File: lxd/patches.go (hunk 1 above)
Comment (roosterfish) [reply lxd/patches.go]: Fixed.

**Review by tomponline — APPROVED** [review]

**Pull request merged**